### PR TITLE
blink: init at 1.0.0

### PR DESCRIPTION
**Pull request by t4ccer**
###### Description of changes

Add blink again.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by totoroot** [comment]
Duplicate of #215404, which is still open.

**Comment by t4ccer** [comment]
Oh sorry, didn't notice that one. Should I close my PR then?

File: pkgs/applications/emulators/blink/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchFromGitHub, lib }:
+
+stdenv.mkDerivation rec {
```
Comment (drupol) [inline pkgs/applications/emulators/blink/default.nix:3]: I think we should use the `finalAttrs` pattern here.

Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

File: pkgs/applications/emulators/blink/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  pname = "blink";
+  version = "0.9.2";
+
+  src = fetchFromGitHub {
+    owner = "jart";
+    repo = pname;
```
Comment (drupol) [inline pkgs/applications/emulators/blink/default.nix:9]:
```suggestion
    repo = "blink";
```

File: pkgs/applications/emulators/blink/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
+  # 'make check' requires internet connection
+  doCheck = true;
+  checkPhase = ''
+    make test
+  '';
+
+  meta = {
+    description = "Tiniest x86-64-linux emulator";
```
Comment (drupol) [inline pkgs/applications/emulators/blink/default.nix:21]:
```suggestion
    description = "Tiniest x86-64-linux emulator";
    longDescription = ''
      blink is a virtual machine that runs x86-64-linux programs on different operating systems and hardware architectures. It's designed to do the same thing as the qemu-x86_64 command, except that
      - blink is much smaller in size than qemu-x86_64
      - blink will run your Linux binaries on any POSIX platform, whereas qemu-x86_64 only supports Linux
      - blink goes 2x faster than qemu-x86_64 on some benchmarks, such as SSE integer / floating point math. Blink is also faster at running ephemeral programs such as compilers
    '';
```

**Review by drupol — CHANGES_REQUESTED** [review]
Hello,

I added a few comments and I took ideas from https://github.com/NixOS/nixpkgs/pull/215404

Can you please update so we can proceed ?

**Comment by t4ccer** [comment]
Blink 1.0 has been just released, let me update before merging

**Comment by t4ccer** [comment]
Well, it just worked

**Review by drupol — APPROVED** [review]

**Pull request merged**